state_machine/applications/system-check/tests/burnwire_test.py

```diff
@@ -0,0 +1,121 @@
+"""
+Python system check script for PyCubed satellite board
+PyCubed Mini mainboard-v02 for Pocketqube Mission
+Burnwire Deployment Test
+* Author(s): Yashika Batra
+"""
+
+import time
+
+# voltage level constants; set between 0 and 1
+v1 = 0.25
+v2 = 0.5
+v3 = 0.75
+
+def user_test(burnnum, burnwire, vlevel):
+    """
+    All user interaction happens in this function
+    Set wait times, change print statement and input formatting, etc.
+    """
+
+    # wait times
+    multimeter_wait_time = 30
+    burn_time = 30
+
+    # other constants
+    projected_voltage = vlevel * 3.3
+    dutycycle = int((vlevel / 100) * (0xFFFF))
+
+    # set up the test
+    print("Testing Burn Wire IC", str(burnnum), "for", projected_voltage, "volts.")
+    print("Please retrieve a multimeter. Waiting", str(multimeter_wait_time), "seconds.")
+    time.sleep(multimeter_wait_time)
+
+    # conduct the test
+    print("Please read the voltage between the ground pin (GND) and burnwire pin on the -Z solar board.")
+    print("Waiting", str(burn_time), "seconds.")
+    burnwire.duty_cycle = dutycycle
+    time.sleep(burn_time)
+    burnwire.duty_cycle = 0
+
+    # gather input and return results
+    print("Burnwire Test Complete")
+    actual_voltage = float(input("Please enter the voltage you recorded: "))
+    return projected_voltage, actual_voltage
+
+
+def voltage_levelx(cubesat, result_dict, vnum, burnnum):
+    """
+    All automation happens in this function
+    Select burnwire and voltage level
+    Process user test results and update the result dictionary accordingly
+    """
+
+    # choose a burnwire or exit
+    burnwire = 0
+    if burnnum == 1:
+        burnwire = cubesat.burnwire1
+    elif burnnum == 2:
+        burnwire = cubesat.burnwire2
+    else:
+        print("Not a valid burnwire.")
+        return result_dict
+
+    # choose a voltage level or exit
+    vlevel = 0
+    if vnum == 1:
+        vlevel = v1
+    elif vnum == 2:
+        vlevel = v2
+    elif vnum == 3:
+        vlevel = v3
+    else:
+        print("Not a valid voltage level option.")
+        return result_dict
+
+    # set string constant
+    result_key = 'Burnwire' + str(burnnum) + '_Volt' + str(vlevel)
+
+    # get user test result values, process and print results
+    projected_voltage, actual_voltage = user_test(burnnum, burnwire, vlevel)
+    result_val_string = ('Burnwire ' + str(burnnum) + ' at voltage level ' +
+                         str(vlevel) + 'returns ' + actual_voltage + ' volts')
+    print(result_val_string)
+
+    # update dictionary values based on user test result values
+    if (abs(projected_voltage - actual_voltage) < 0.2):
```

Comment:
Honestly, I have no idea how close these should actually be. Rather than predicting the voltage, you could just have the user indicate if the burn wire functions properly.

Reply:
I remember when Neil and I tested it last, it was within 0.1 volts (which is why I expanded the range to 0.2). However, you are correct that this is something that's a little more unpredictable -- I can just ask for a Y/N input from the user and log the result of that if you prefer.